### DISCUSSION TITLE
Celery 5.3.4: Memory leak in Celery master process

### DISCUSSION THREAD

**Discussion by narva8916** [celery/celery#8514] [Q&A]
Recently we upgrade from celery 4.4.3 to 5.3.4 in order to overcome a memory leak issue we were experiencing.

After that we see there's a minimum change about ram consumption but anyway the workers still consuming ram even when no task are executed.

This is the ram memory consumption from one of the workers, from 15 to 18 there were no task executed, anyway the worker is consuming memory.

![Screen Shot 2023-09-18 at 10 16 29 AM](https://github.com/celery/celery/assets/3817743/1bac2df5-efb9-46b0-9b95-14f9d6536bad)

We are using celery 5.3.4 and python 3.9.0.

Someone have experience this same issue? how do you fix it?

**Comment by auvipy** [comment]
you have to use memory leak tool to find out the leaking objects/roots of that

**Comment by thedrow** [comment]
Can you try profiling with [memray](https://github.com/bloomberg/memray) to see where the memory leak is coming from?

**Comment by esauser** [comment]
We're seeing a similar issue.  See [this](https://github.com/celery/celery/issues/8521#issuecomment-2016922912) comment.
